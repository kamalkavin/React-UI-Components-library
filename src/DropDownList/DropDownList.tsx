/**
 *
 * DropDownList
 *
 */

import React, { FC } from 'react';
import { Dropdown } from 'react-bootstrap';
import SvgIcon from 'components/SvgIcon/SvgIcon';
import { IconList } from 'components/SvgIcon/SVGIconLibrary';

interface IProps {
	label?: string;
	list: { value: string; label: string; icon?: IconList; size?: number; color?: string }[];
	onSelect(e: string): void;
	selected?: string;
	className?: string;
	withIcon?: boolean;
	viewLabel?: boolean;
}

const DropDownList: FC<IProps> = ({ label, list, onSelect, selected, className = '', withIcon = false, viewLabel = true  }: IProps) => {
	const selectedItem = list.find(item => item.value === selected)?.label;
	const selectedIcon = list.find((item) => item.value === selected);

	return (
		<div className={`dropdown-list ${className}`} id='drop-down-list' data-cy="view-by-dropdown">
			<Dropdown className='pd-container'>
				<Dropdown.Toggle className='pd-btn'>
					{label && <span className='pd-lbl'>{label}</span>}
					{viewLabel && selectedItem}
					{!viewLabel && selectedIcon && <SvgIcon icon={selectedIcon.icon || 'Settings'} size={selectedIcon.size || 20} color={selectedIcon.color}/>}
					<span className='pd-arrow'>
						<SvgIcon size={16} icon='down-arrow' className='navbar-warn-img' />
					</span>
				</Dropdown.Toggle>

				<Dropdown.Menu>
					{list.map((item, i) => (
						<Dropdown.Item
							data-cy={item.label}
							key={`dropdown-list${i}`}
							href={item.label}
							disabled={item.value === selected}
							onClick={e => {
								e.preventDefault();
								onSelect(item.value);
								document.body.click();
							}}>
							{viewLabel && item.label}
							{withIcon && <SvgIcon icon={item.icon|| 'Settings'} size={item.size || 0} color={item.color}/>}
						</Dropdown.Item>
					))}
				</Dropdown.Menu>
			</Dropdown>
		</div>
	);
};

export default DropDownList;
