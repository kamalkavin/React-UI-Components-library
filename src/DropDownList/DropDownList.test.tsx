/**
 * Tests for DropDown Button
 */
import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import DropDownList from './DropDownList';
import '@testing-library/jest-dom/extend-expect';

describe('DropDown button container', () => {
	afterEach(() => {
		cleanup();
	});
	test('Dropdown button test case', () => {
		const list = [
			{ value: 'correlated', label: 'Correlated Alarms' },
			{ value: 'alarms', label: 'All Alarms' },
		];
		const viewBy = 'alarms';
		render(
			<DropDownList
				label='View By: '
				className='criteria-pill-container'
				selected={viewBy}
				list={list}
				onSelect={(value) => {
					document.body.innerHTML = '<span data-cy="dropdwon-selected"></span>';
				}}
			/>
		);
		expect(screen.getByTestId('view-by-dropdown')).toBeInTheDocument();
		fireEvent.click(screen.getByText(/All Alarms/i));
		expect(screen.getByTestId(list[0].label)).toBeInTheDocument();
		expect(screen.getByTestId(list[1].label)).toBeInTheDocument();
		fireEvent.click(screen.getByTestId(list[0].label));
		expect(screen.getByTestId('dropdwon-selected')).toBeInTheDocument();
	});
	test('Dropdown button test case', () => {
		const list = [
			{ value: 'correlated', label: 'Correlated Alarms' },
			{ value: 'alarms', label: 'All Alarms' },
		];
		const viewBy = 'correlated';
		render(
			<DropDownList
				label='View By: '
				selected={viewBy}
				list={list}
				onSelect={(value) => {
					document.body.innerHTML = '<span data-cy="dropdwon-selected"></span>';
				}}
			/>
		);
		expect(screen.getByTestId('view-by-dropdown')).toBeInTheDocument();
		fireEvent.click(screen.getByText(/Correlated Alarms/i));
		expect(screen.getByTestId(list[0].label)).toBeInTheDocument();
		expect(screen.getByTestId(list[1].label)).toBeInTheDocument();
		fireEvent.click(screen.getByTestId(list[1].label));
		expect(screen.getByTestId('dropdwon-selected')).toBeInTheDocument();
	});
});