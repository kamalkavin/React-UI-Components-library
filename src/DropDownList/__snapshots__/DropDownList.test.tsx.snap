// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownList Snapshot renders correctly 1`] = `
<div
  className="dropdown-list "
  data-cy="view-by-dropdown"
  id="drop-down-list"
>
  <Dropdown
    className="pd-container"
    navbar={false}
  >
    <DropdownToggle
      className="pd-btn"
    >
      <span
        className="pd-arrow"
      >
        <DownArrow />
      </span>
    </DropdownToggle>
    <DropdownMenu
      align="left"
      alignRight={false}
      flip={true}
    >
      <DropdownItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        data-cy=""
        disabled={false}
        href=""
        key="dropdown-list0"
        onClick={[Function]}
      />
    </DropdownMenu>
  </Dropdown>
</div>
`;
